yt_dlp/extractor/vrt.py

```diff
@@ -229,102 +319,104 @@ class VrtNUIE(VRTBaseIE):
-
-        watch_info = traverse_obj(details, (
-            'actions', lambda _, v: v['type'] == 'watch-episode', {dict}), get_all=False) or {}
-        video_id = join_nonempty(
-            'episodePublicationId', 'episodeVideoId', delim='$', from_dict=watch_info)
-        if '$' not in video_id:
-            raise ExtractorError('Unable to extract video ID')
-
-        vrtnutoken = self._download_json(
-            'https://token.vrt.be/refreshtoken', video_id, note='Retrieving vrtnutoken',
-            errnote='Token refresh failed')['vrtnutoken'] if self._authenticated else None
-
-        video_info = self._call_api(video_id, 'vrtnu-web@PROD', vrtnutoken)
-
-        if 'title' not in video_info:
-            code = video_info.get('code')
-            if code in ('AUTHENTICATION_REQUIRED', 'CONTENT_IS_AGE_RESTRICTED'):
-                self.raise_login_required(code, method='password')
-            elif code in ('INVALID_LOCATION', 'CONTENT_AVAILABLE_ONLY_IN_BE'):
-                self.raise_geo_restricted(countries=['BE'])
-            elif code == 'CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS_AND_EXPATS':
-                if not self._authenticated:
-                    self.raise_login_required(code, method='password')
-                self.raise_geo_restricted(countries=['BE'])
-            raise ExtractorError(code, expected=True)
-
-        formats, subtitles = self._extract_formats_and_subtitles(video_info, video_id)
+
+        # 1. Obtain/refresh 'vrtnu-site_profile' tokens
+        # 1.a Visit 'authorize' URL again
+        headers = {
+            'Cookie': self._cookies.header(['OIDCXSRF', 'SESSION'])
+        }
+        request = urllib.request.Request(self._auth_url, headers=headers)
+        res = self._opener.open(request, None)
+        callback_url = res.headers.get_all('Location')[0]
+
+        # 1.b Follow redirection: visit 'callback' URL
+        headers = {
+            'Cookie': self._cookies.header(['oidcstate'])
+        }
+        request = urllib.request.Request(callback_url, headers=headers)
+        res = self._opener.open(request, None)
+
+        # 2. Perform GraphQL query to obtain video metadata
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {self._cookies["vrtnu-site_profile_at"]}'
+        }
+
+        data = {
+            'operationName': 'VideoPage',
+            'query': self._VIDEOPAGE_QUERY,
+            'variables': {
+                'pageId': f'{parsed_url.path.rstrip("/")}.model.json'
+            }
+        }
+
+        metadata = self._download_json(
+            'https://www.vrt.be/vrtnu-api/graphql/v1',
+            display_id, 'Downloading asset JSON', 'Unable to download asset JSON', headers=headers, data=json.dumps(data).encode())['data']['page']
+
+        video_id = metadata['episode']['watchAction']['streamId']
+        title = metadata['seo']['title']
+        season_number = int(metadata['episode']['onTimeRaw'][:4])
```

Comment:
This should be non-fatal and inlined. Just keep it as part of the traverse_obj call.

Reply:
Done. But I had to create a 'parse_year()' function that extracts the first 4 characters before calling 'int_or_none()'.